src/components/TextInput/v1/TextInput.md

```diff
@@ -0,0 +1,294 @@
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" isDark />
+  </div>
+</div>
+```
+
+###### Input states
+Text input's have 8 basic states.
+ - Idle unfilled
+ - Idle filled
+ - Focused unfilled
+ - Focused filled
+ - Saved<sup>*</sup> *not yet implamented*
+ - Valid
+ - Invalid
+ - Read Only / Disabled
+
+Idle Unfilled / Focused Unfilled
+
+When a user has clicked or pressed into an input it’s in focused state. 
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" isDark />
+  </div>
+</div>
+```
+
+Idle Filled / Focused Filled
+
+When a field has been filled in by the user, and has been unfocused it shows idle and filled in. When a user has filled in a text input and clicked back into the text field it is shown as focused and has an icon button that allows the user to quickly clear the whole field and edit it from scratch.
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" isDark />
+  </div>
+</div>
+```
+
+Valid
+
+When information is required and/or needs to be formatted in a specific way the text input should be validated. When it’s successful the field shows a success state. Passing the `hasBeenValidated` prop to the input will enabled the valid state.
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" hasBeenValidated />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" hasBeenValidated isDark />
+  </div>
+</div>
+```
+Invalid
+
+When information is required and/or needs to be formatted in a specific way the text input should be validated. When a input can’t be validated the input is highlighted with error styling. Passing an `errors` array to the input to enabled the invalid state.
```

Comment:
"Passing an `errors` array"

change to

"Pass a non-empty `errors` array"

And add this sentence after:

Typically an ancestor form component will do the work of setting `errors` properly for you.